### PR TITLE
KubeVirt: update migration guide to 2.22 and switch to instancetype

### PR DESCRIPTION
**Pull request by hdurand0710**
#Fixes: kubermatic/docs#1261
Signed-off-by: Helene Durand <helene@kubermatic.com>

### CODE REVIEW

File: content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 1]
```diff
@@ -512,3 +654,125 @@ Follow the below steps to import the dashboard in Grafana:
...
+{{% /notice %}}
+
+
+
+### Features that need some manual migration of the `MachineDeployment`:
+
+List of deprecated/updated features that require some manual update in the `MachineDeployment`:
+
+
+| Topic                      | Deprecated /Updagred                                          | In Favor of               | Mandatory/Optional migration |
```
Comment (mfranczy) [inline content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md:675]: Please change to upgraded :)

Comment (hdurand0710) [reply content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md]: Done

File: content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 2]
```diff
@@ -344,85 +383,188 @@ spec:
...
-- `Devices.Watchdog`
-- `IOThreadsPolicy`
+#### How to provide your own **Custom** instancetypes and preferences
 
-*Note3:* Migration to the instanceType API
 
-Please note that in the next KKP release, we will migrate to the new *VirtualMachineInstancetype* ([alpha release](https://github.com/kubevirt/api/blob/main/instancetype/v1alpha1/types.go#L35)),
-as the *VirtualMachineInstancePreset* will be deprecated. This will for example lift the limitation around the list of
-merged fields from the `VirtualMachineInstance` and provide deterministic behavior with immediate detection of merge conflicts.
+Just create some `VirtualMachinClusterInstancetype` and/or some `VirtualMachineClusterPreference` (custer-wide resources) in the KubeVirt infrastructure cluster and you are can use them to template your VMs.
```
Comment (sankalp-r) [inline content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md:567]: small typo.
`custer-wide` should be `cluster-wide`

File: content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 3]
```diff
@@ -512,3 +654,125 @@ Follow the below steps to import the dashboard in Grafana:
...
+
+
+| Topic                      | Deprecated /Upgraded                                          | In Favor of               | Mandatory/Optional migration |
+|----------------------------|------------------------------------------------------|---------------------------|-------|
+| Virtual Machine Scheduling | Deprecated: `Pod Affinity Preset` and `Pod Anti Affinity Preset` | `TopologySpreadConstraints` | **Optional**: only if your `MachineDeployment` did contain `podAffinityPreset` or `podAntiAffinityPreset` |
+| Virtual Machines Templating | Deprecated: `Flavor`            | `Instancetype` and `Preference`           | **Optional**: only if your `MachineDeployment` did contain `flavor` |
+| Upgrade of [KubeVirt CCM](https://github.com/kubevirt/cloud-provider-kubevirt) from v0.2.0 to v0.4.0 | Upgrade Kubevirt CCM version            | (needed for LoadBalancer services)        | **Mandatory**: all existing `MachineDeployment` must be updated |
+
+{{% notice warning %}}
+**Perform all the needed migration of you existing `MachineDeployment` following the below procedure according to each topic that needs migration.**
```
Comment (sankalp-r) [inline content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md:682]: typo.
s/you/your

File: content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 4]
```diff
@@ -512,3 +654,125 @@ Follow the below steps to import the dashboard in Grafana:
...
+Updating the `MachineDeployment` will create a new `VirtualMachine`  - Perform all the needed changed at once.
```
Comment (sankalp-r) [inline content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md:683]: `changed` should be `change`, right?

File: content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md (hunk 2 above)
Comment (hdurand0710) [reply content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md:567]: done

File: content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md (hunk 4 above)
Comment (hdurand0710) [reply content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md:683]: correct. Fixed.

File: content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md (hunk 3 above)
Comment (hdurand0710) [reply content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md]: Done

File: content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 5]
```diff
@@ -512,3 +654,125 @@ Follow the below steps to import the dashboard in Grafana:
 ## Breaking Changes
 
 Please be aware that between KKP 2.20 and KKP 2.21, a breaking change to the `MachineDeployment` API for KubeVirt has occurred. For more details, please [check out the 2.20 to 2.21 upgrade notes]({{< ref "../../../tutorials-howtos/upgrading/upgrade-from-2.20-to-2.21/#kubevirt-migration" >}}).
+
+## Migration from KKP 2.21 to KKP 2.22
+
+{{% notice note %}}
+Note that the Vms (and their workload) created from `MachineDeployments` with KKP 2.21 are not affected by the migration from KKP 2.21 to KKP 2.22 **if no update is done in the `MachineDeployment`**. You can safely perform the *MachineDeployment* manual migration after KKP upgrade from 2.21 to 2.22. `MachineDeployments` (and associated VM/VMis) do not require any action before the migration from KKP 2.21 to KKP 2.22.
```
[the remark this answers is not in the document]
Comment (hdurand0710) [reply content/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt.en.md]: done

**Pull request merged**